src/scripts/run_certbot.sh

```diff
@@ -87,6 +114,14 @@ for conf_file in /etc/nginx/conf.d/*.conf*; do
             key_type="rsa"
         fi
 
+        # Determine the authenticator to use to solve the challenge
+        if [[ "${cert_name,,}" =~ ^.*(-|\.)(dns-($(echo ${CERTBOT_DNS_AUTHENTICATORS} | sed 's/ /|/g'))).*$ ]]; then
+            challenge_authenticator=${BASH_REMATCH[2]}
+            debug "Found mention of authenticator '${challenge_authenticator}' in name '${cert_name}'"
+        else
+            challenge_authenticator="${CERTBOT_DEFAULT_AUTHENTICATOR:-webroot}"
```

Comment:
I would like to squeeze in something like this here as well

```
if [ -n "${CERTBOT_DEFAULT_AUTHENTICATOR}" ] || [[ "${CERTBOT_DEFAULT_AUTHENTICATOR,,}" == webroot ]]; then 
     challenge_authenticator="webroot"
else 
    error "Unknown authenticator '${CERTBOT_DEFAULT_AUTHENTICATOR}'"
    exit 1
fi
```

Reply:
Do we want to `exit 1` or skip to next cert ?

Also, slightly confused by the logic here?

- `if DEFAULT_AUTHENTICATOR is not empty || DEFAULT_AUTHENTICATOR is webroot`
   Redundant, did you mean -z instead of -n ?

- `else error unknown authenticator DEFAULT_AUTHENTICATOR`
   Do you want to check that value against the authenticators list ?

Reply:
So I kept the logic simple here, but made the full check in `get_certificates`, where we actually check that the authenticator we got as parameter exists!